src/dmd/mars.d

```diff
@@ -1499,6 +1499,42 @@ bool parseCommandLine(const ref Strings arguments, const size_t argc, ref Param
             params.useDeprecated = Diagnostic.inform;
         else if (arg == "-c")                // https://dlang.org/dmd.html#switch-c
             params.link = false;
+        else if (startsWith(p + 1, "check=")) // https://dlang.org/dmd.html#switch-check
+        {
+            /* Parse:
+             *    -check=[assert|bounds|in|invariant|out|switch][=[on|off]]
+             */
+
+            // Check for legal option string; return true if so
+            static bool check(const(char)* p, string name, ref CHECKENABLE ce)
```

Comment:
This seems awfully rigid, re checking for exactly `=on` or `=off` 
For a first implementation it might be alight though.